controllers/machine.go

```diff
@@ -330,6 +336,26 @@ func (mrc *machineReconcileContext) ensureHardwareUserData(hardware *tinkv1.Hard
 	return nil
 }
 
+func (mrc *machineReconcileContext) ensureHardwareDiskData(hardware *tinkv1.Hardware) error {
+	if hardware.Spec.Disks == nil {
+		foundDisks := make([]tinkv1.Disk, 0)
+
+		if hardware.Spec.Metadata != nil &&
+			hardware.Spec.Metadata.Instance != nil &&
+			hardware.Spec.Metadata.Instance.Storage != nil {
+			for _, disk := range hardware.Spec.Metadata.Instance.Storage.Disks {
+				foundDisks = append(foundDisks, tinkv1.Disk{Device: disk.Device})
+			}
+		} else {
+			return ErrPatchingHardwareDiskData
+		}
+
+		hardware.Spec.Disks = foundDisks
+	}
+
+	return nil
+}
```

Comment:
Why is this needed here? This seems like something that should be handled by tink-controller and not CAPT?